addons/survey/models/survey_user_input.py

```diff
@@ -786,24 +786,35 @@ def _check_answer_type_skipped(self):
 
     @api.model_create_multi
     def create(self, vals_list):
-        for vals in vals_list:
+        # Setup prefetching
+        suggested_answers = self.env['survey.question.answer'].browse(vals.get('suggested_answer_id', False) for vals in vals_list)
+        questions = self.env['survey.question'].browse(vals.get('question_id', False) for vals in vals_list)
+        user_inputs = self.env['survey.user_input'].browse(vals.get('user_input_id', False) for vals in vals_list)
+        if any(u.survey_id.session_speed_rating and u.is_session_answer for u in user_inputs):
+            existing_user_inputs = self.env['survey.user_input'].search([('id', 'in', user_inputs.ids)])
+            user_inputs = [user_input if user_input in existing_user_inputs else False for user_input in user_inputs]
```

Comment:
Is this part really worth it VS the added complexity?

From what I understand, we prefetch user inputs, then we fetch all of their surveys (seems a bit counter-intuitive when trying to improve performances, but I'm guessing "_get_answer_score_values" does that further down the line), then check some conditions to do some extra processing.

That extra processing searches for user inputs, then changes the browserecord to a list based on existence comparison?

Which means you are zipping a mix of recordset and lists here below?

What are we gaining here exactly?

Reply:
To be complete as it is still part of the new version: this was about checking once for the existence of all referenced user_inputs in flows that should count a speed adjusted score (instead of in a loop).